dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -1120,4 +1120,17 @@ public Response getToolZip(@ApiParam(hidden = true) @Auth Optional<User> user,
         return Response.ok().entity((StreamingOutput)output -> writeStreamAsZip(sourceFiles, output, path))
             .header("Content-Disposition", "attachment; filename=\"" + fileName + "\"").build();
     }
+
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("{alias}/aliases")
+    @ApiOperation(value = "Retrieves a tool by alias.", response = Tool.class)
+    public Tool getToolByAlias(@ApiParam(value = "Alias", required = true) @PathParam("alias") String alias) {
+        final Tool tool = this.toolDAO.findByAlias(alias);
```

Comment:
Do you not have to check if the tool is published? And if so, does that mean you should add an optional User?

Reply:
Yes good find, I'll change it to use optional oauth.